__tests__/server/middleware/sendHtml.spec.js

```diff
@@ -825,7 +839,133 @@ describe('sendHtml', () => {
       expect(res.send.mock.calls[0][0]).not.toMatch('[object ');
       expect(res.send.mock.calls[0][0]).not.toContain('undefined');
     });
+    it('returns default error page if fetching custom error page url fails', async () => {
+      const errorPageUrl = 'https://example.com';
+      const fetchError = new Error('getaddrinfo ENOTFOUND');
+      global.fetch = jest.fn(() => Promise.reject(fetchError));
+
+      await setErrorPage(errorPageUrl);
+      renderStaticErrorPage(res);
+
+      expect(res.send).toHaveBeenCalledTimes(1);
+      expect(res.send.mock.calls[0][0]).toContain('<!DOCTYPE html>');
+      expect(res.send.mock.calls[0][0]).toContain('<meta name="application-name" content="one-app">');
+    });
+    it('returns provided error page if provided', async () => {
+      const errorPageUrl = 'https://example.com';
+      const mockResponse = `<!doctype html>
+      <html>
+      <head>
+        <title>Custom Error Page</title>
+        <meta charset="utf-8" />
+        <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+      </head>
+      <body>
+        <div>
+          <h1>Custom Error Page</h1>
+          <p>This is a custom error page.</p>
+        </div>
+      </body>
+      </html>`;
+
+      global.fetch = jest.fn(() => Promise.resolve({
+        text: () => Promise.resolve(mockResponse),
+        headers: new global.Headers({
+          'Content-Type': 'text/html',
+        }),
+      }));
+
+      await setErrorPage(errorPageUrl);
+      renderStaticErrorPage(res);
+
+      const data = await global.fetch.mock.results[0].value;
+
+      expect(global.fetch).toHaveBeenCalledTimes(1);
+      expect(global.fetch).toHaveBeenCalledWith(errorPageUrl);
+      expect(await data.text()).toBe(mockResponse);
+      expect(await data.timeout).toBe(6000);
+      expect(res.send).toHaveBeenCalledTimes(1);
+      expect(res.send.mock.calls[0][0]).toContain('<!doctype html>');
+      expect(res.send.mock.calls[0][0]).toContain('<h1>Custom Error Page</h1>');
+    });
   });
+
+  describe('setErrorPage', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    const errorPageUrl = 'https://example.com';
+    const mockResponse = `<!doctype html>
+  <html>
+  <head>
+    <title>Custom Error Page</title>
+    <meta charset="utf-8" />
+    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+  </head>
+  <body>
+    <div>
+      <h1>Custom Error Page</h1>
+      <p>This is a custom error page.</p>
+    </div>
+  </body>
+  </html>`;
+    it('fetches errorPageUrl', async () => {
```

Comment:
This tests it is very similar to `returns provided error page if provided` above. I know we are testing the `setErrorPage` function and the `renderStaticErrorPage` separately but maybe gather some more feedback from the team to see if it is required

Reply:
So this tests `setErrorPage` which gets called during `onModuleLoad`, the other test is testing that `renderStaticErrorPage` is actually sending that custom error page. I believe there should be two separate tests.